airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/GCSDestination.java

```diff
@@ -0,0 +1,73 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.gcs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.integrations.BaseConnector;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.destination.jdbc.copy.gcs.GcsConfig;
+import io.airbyte.integrations.destination.jdbc.copy.gcs.GcsStreamCopier;
+import io.airbyte.integrations.destination.gcs.writer.ProductionWriterFactory;
+import io.airbyte.integrations.destination.gcs.writer.GCSWriterFactory;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GCSDestination extends BaseConnector implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GCSDestination.class);
+
+  public static void main(String[] args) throws Exception {
+    new IntegrationRunner(new GCSDestination()).run(args);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      GcsStreamCopier.attemptGcsWriteAndDelete(GcsConfig.getGcsConfig(config));
+      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+    } catch (Exception e) {
+      LOGGER.error("Exception attempting to access the GCS bucket: ", e.toString());
+      return new AirbyteConnectionStatus()
+          .withStatus(AirbyteConnectionStatus.Status.FAILED)
+          .withMessage("Could not connect to the GCS bucket with the provided configuration. \n" + e
+              .getMessage());
+    }
+  }
+
+  @Override
+  public AirbyteMessageConsumer getConsumer(JsonNode config,
+                                            ConfiguredAirbyteCatalog configuredCatalog,
+                                            Consumer<AirbyteMessage> outputRecordCollector) {
+    GCSWriterFactory formatterFactory = new ProductionWriterFactory();
+    return new GCSConsumer(GCSDestinationConfig.getGCSDestinationConfig(config), configuredCatalog, formatterFactory, outputRecordCollector);
+  }
+}
```

Comment:
Nit. Missing newline.

Reply:
you mean between line 72 and line 73? if so, fixed

Reply:
Oh, it's not between line 72 and 73. It's at the end of this file. you can see the symbol as a reminder from GitHub:

<img width="102" alt="Screen Shot 2021-06-25 at 07 51 23" src="https://user-images.githubusercontent.com/1933157/123442913-3cf7ad80-d58a-11eb-98d8-b6514e0b0233.png">

This is a nitpick. The argument is [here](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).

Reply:
Interesting read! Thanks for it :)
fixed now btw